File: servers/PatientService/Models/PatientReport.cs
using System.ComponentModel.DataAnnotations;

namespace PatientService.Models
{
    public class PatientReport
    {
        [Key]
        public string Id { get; set; } = string.Empty;

        public string? Title { get; set; }
        
        public string? Content { get; set; } // markdown content

        public string? PatientEmail { get; set; } // is user's email

        public bool IsSent { get; set; } = false; // if IsSent is true => doctor cannot edit the report

        public DateTime? SentDate { get; set; }

        public bool IsRead { get; set; } = false;

        public string? SentById { get; set; }

        public string? SentByFullname { get; set; }
    }
}

File: mobile_app/lib/features/model/patient_report/patient_report.dart
class PatientReport {
  final String id;
  final String? title;
  final String? content;
  final String? patientEmail;
  final bool isSent;
  final String? sentDate;
  final bool isRead;
  final String? sentById;
  final String? sentByFullname;

  PatientReport({
    required this.id,
    this.title,
    this.content,
    this.patientEmail,
    required this.isSent,
    this.sentDate,
    required this.isRead,
    this.sentById,
    this.sentByFullname,
  });

  factory PatientReport.fromJson(Map<String, dynamic> json) {
    return PatientReport(
      id: json['id'] ?? '',
      title: json['title'],
      content: json['content'],
      patientEmail: json['patientEmail'],
      isSent: json['isSent'] ?? false,
      sentDate: json['sentDate'],
      isRead: json['isRead'] ?? false,
      sentById: json['sentById'],
      sentByFullname: json['sentByFullname'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'content': content,
      'patientEmail': patientEmail,
      'isSent': isSent,
      'sentDate': sentDate,
      'isRead': isRead,
      'sentById': sentById,
      'sentByFullname': sentByFullname,
    };
  }
}
